### PR TITLE
re-generate rtdb's api.txt and fail build if an error occurs parsing the file

### PR DESCRIPTION
**Pull request by rosariopfernandes**
The api.txt file should have been regenerated in #1672, but the [build](https://android-ci.firebaseopensource.com/view/gcs/android-ci/pr-logs/pull/firebase_firebase-android-sdk/1672/api-information/1273978984790495232/) failed silently with a warning:

```
Unable to parse signature file /home/prow/go/src/github.com/firebase/firebase-android-sdk/firebase-database/api.txt:

 /home/prow/go/src/github.com/firebase/firebase-android-sdk/firebase-database/api.txt:191:

 expected ; found (
```

This PR should:
- unset `setIgnoreExitValue(true)` to prevent the build from failing silently;
- regenerate the firebase-database's api.txt file.

### CODE REVIEW

**Comment by samtstern** [comment]
@rosariopfernandes didn't realize you were already working on this!  No need to base this on my PR, you can base this on `master`.  This change is good.

You can just copy the `@hide` comment from #1690 (just pushed it)

**Comment by rosariopfernandes** [comment]
@samtstern Seems like I was also seeing an outdated thread and I only saw the recent comments when I posted mine.

I'll base this on master and add the `@hide` comment

File: firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java [hunk 1]
```diff
@@ -215,6 +233,23 @@ private void ensureClientConfigured() {
     }
   }
 
+  private FirebaseFirestoreSettings mergeEmulatorSettings(
```
Comment (samtstern) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java:236]: Uh oh, now all my code leaked in here :-/

Comment (rosariopfernandes) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java:236]: Oops, will fix!

Comment (rosariopfernandes) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java:236]: @samtstern for some reason I can't revert that commit locally. I'll close this PR and open a clean one

Comment (samtstern) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java:236]: @rosariopfernandes you can also just force-push onto this same branch, GitHub will handle it.

Comment (rosariopfernandes) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java]: @samtstern done! Thanks for the tip

**Comment by samtstern** [comment]
Just gonna have @rlazo double-check this because I am not a normal SDK approver, although this LGTM!

**Comment by rlazo** [comment]
Done, thanks!

**Pull request merged**